public/html/alerts/EditAlertViewModel.html

```diff
@@ -18,54 +18,44 @@
-                    </div>
-                    <div class="form-group">
-                        <label for="serviceInput">Service</label>
-                        <input class="form-control" data-bind="value: service" type="text" id="serviceInput" placeholder="Service">
-                    </div>
-                    <div class="form-group">
-                        <label for="clusterInput">Cluster</label>
-                        <input class="form-control" data-bind="value: cluster" type="text" id="clusterInput" placeholder="Cluster">
-                    </div>
-                    <div class="form-group">
-                        <label for="statisticInput">Statistic</label>
-                        <input class="form-control" data-bind="value: statistic" type="text" id="statisticInput" placeholder="Statistic">
-                    </div>
-                    <div class="form-group">
-                        <label for="periodInput">Period</label>
-                        <input class="form-control" data-bind="value: period" type="text" id="periodInput" placeholder="Period">
-                    </div>
-                    <div class="form-group">
-                        <label for="operatorInput">Operator</label>
-                        <select class="form-control" data-bind="options: operators, optionsText: 'text', optionsValue: 'value', value: operator" id="operatorInput"></select>
-                        <!--<input class="form-control" data-bind="value: operator" type="text" id="operatorInput" placeholder="Operator">-->
-                    </div>
-                    <div class="form-group">
-                        <label for="valueInput">Value</label>
-                        <input class="form-control" data-bind="value: value" type="text" id="valueInput" placeholder="Value">
+                    <div class="form-group" data-bind="css: { 'has-error': queryErrors().length > 0, 'has-warning': queryWarnings().length > 0 }">
+                        <label for="queryInput">Query</label>
+                        <textarea class="form-control" data-bind="textInput: query" rows="3" id="queryInput" placeholder="SELECT ..."></textarea>
+                        <!-- ko foreach: queryErrors -->
+                        <span class="help-block" data-bind="text: $data"></span>
+                        <!-- /ko -->
+                        <!-- ko foreach: queryWarnings -->
+                        <span class="help-block" data-bind="text: $data"></span>
+                        <!-- /ko -->
                     </div>
                     <div class="form-group">
-                        <label for="unitInput">Unit</label>
-                        <input class="form-control" data-bind="value: valueUnit" type="text" id="unitInput" placeholder="Unit">
+                        <label for="periodInput">Check Interval</label>
```

Comment:
Ahh now I understand what period is ;)

The problem is that this assumes that the query execution model is always periodic. It precludes stream based evaluation or other callback execution models. Not saying we should support it, but want to make sure you're making that call explicitly.

Also, we should rename it from `period` to `checkInterval`.

Reply:
Changed.